#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import (ParameterGenerator,
                                                            int_t)

PACKAGE = "robotics_hw1"

gen = ParameterGenerator()

odometry_integration_method_enum = gen.enum(
    [
        gen.const("Euler", int_t, 0, "Use the Euler integration method"),
        gen.const(
            "RungeKutta",
            int_t,
            1,
            "Use the Runge-Kutta integration method",
        ),
    ],
    "The enum with the integration method to compute the odometry with",
)

gen.add(
    "odometry_integration_method",
    int_t,
    0,
    "The integration method to compute the odometry with",
    0,
    edit_method=odometry_integration_method_enum,
)

exit(gen.generate(PACKAGE, "robotics_hw1", "OdometryCalculator"))
